### PR TITLE
HEEDLS-532 Make filters in the same category OR not AND

### PR DESCRIPTION
**Pull request by AlexJacksonDS**
Changed filters so that filters in the same group work like an OR rather than an AND.

### CODE REVIEW

File: DigitalLearningSolutions.Web/Helpers/FilteringHelper.cs [hunk 1]
```diff
@@ -20,70 +20,106 @@ public static class FilteringHelper
...
+        private class AppliedFilter
+        {
+            public AppliedFilter(string filterOption)
+            {
+                var splitFilter = filterOption.Split(Separator);
+                Group = splitFilter[0];
+                PropertyName = splitFilter[1];
+                PropertyValue = splitFilter[2];
+            }
+            public string Group { get; }
```
Comment (DanBloxham-sw) [inline DigitalLearningSolutions.Web/Helpers/FilteringHelper.cs:57]: The FilterOptions defined below define this as "Category". Should this be the same for consistency or am I misunderstanding what it represents?

**Review by DanBloxham-sw — APPROVED** [review]
Looks good to me. Minor comment about a property name, but its good either way.

File: DigitalLearningSolutions.Web/Helpers/FilteringHelper.cs [hunk 2]
```diff
@@ -20,70 +20,106 @@ public static class FilteringHelper
...
+            string propertyName,
+            string propertyValueString
+        )
+        {
+            var propertyType = typeof(T).GetProperty(propertyName)!.PropertyType;
+            var propertyValue = TypeDescriptor.GetConverter(propertyType).ConvertFromString(propertyValueString);
+            return items.Where(propertyName, propertyValue);
+        }
+
+        private class AppliedFilter
```
Comment (stellake) [inline DigitalLearningSolutions.Web/Helpers/FilteringHelper.cs:48]: Looking at this, I think this file will get quite big quite quickly. Could we move this class to models and/or separate out the filteroptions?

File: DigitalLearningSolutions.Web.Tests/Helpers/FilteringHelperTests.cs [hunk 3]
```diff
@@ -32,7 +32,21 @@ public void FilterItems_returns_expected_items_with_multiple_filters()
...
+
+            // When
+            var result = FilteringHelper.FilterItems(InputItems, filterBy);
+
+            // Then
+            result.Should().BeEquivalentTo(expectedItems);
+        }
+
+        [Test]
+        public void FilterItems_with_filters_in_same_category_returns_union_of_filters()
```
Comment (stellake) [inline DigitalLearningSolutions.Web.Tests/Helpers/FilteringHelperTests.cs:45]: I think we should have a few extra tests for this functionality:
* Filtering two different categories not taking union
* Filtering twice per category not returning duplicates if one item matches both filters in that category
* Case where we test it all together - e.g. two filters in same category and one in a different category

File: DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Centre/Administrator/CentreAdministratorsViewModel.cs [hunk 4]
```diff
@@ -50,7 +50,11 @@ int page
             Admins = paginatedItems.Select(adminUser => new SearchableAdminViewModel(adminUser));
             IEnumerable<FilterOptionViewModel> categoryOptions =
                 categories.Select(
-                    c => new FilterOptionViewModel(c, $"{nameof(AdminUser.CategoryName)}|{c}", FilterStatus.Default)
+                    c => new FilterOptionViewModel(
+                        c,
+                        $"{nameof(AdminUser.CategoryName)}|{nameof(AdminUser.CategoryName)}|{c}",
```
Comment (stellake) [inline DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Centre/Administrator/CentreAdministratorsViewModel.cs:55]: Should we use the "Separator" reference here?

File: DigitalLearningSolutions.Web.Tests/Helpers/FilteringHelperTests.cs (hunk 3 above)
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Web.Tests/Helpers/FilteringHelperTests.cs:45]: Filters in 2 categories already had a test FilterItems_returns_expected_items_with_multiple_filters

File: DigitalLearningSolutions.Web.Tests/Helpers/FilteringHelperTests.cs [hunk 5]
```diff
@@ -32,7 +32,49 @@ public void FilterItems_returns_expected_items_with_multiple_filters()
...
+
+            // When
+            var result = FilteringHelper.FilterItems(InputItems, filterBy);
+
+            // Then
+            result.Should().BeEquivalentTo(expectedItems);
+        }
+
+        [Test]
+        public void FilterItems_with_filters_in_same_category_returns_no_duplicates()
```
Comment (stellake) [inline DigitalLearningSolutions.Web.Tests/Helpers/FilteringHelperTests.cs:59]: in the same group?

**Review by stellake — APPROVED** [review]
Looks good - just need to modify the test names 👍  No need for rereview though

**Pull request merged**